File: frontend-react/src/services/pillowService.ts
import api, { ENDPOINTS } from '../config/api';
import type { Pillow, DefaultSize, PageResponse } from '../types';

export const pillowService = {
  getAll: async (page = 0, size = 12): Promise<PageResponse<Pillow>> => {
    const response = await api.get(ENDPOINTS.PILLOWS, { params: { page, size, sort: 'id,desc' } });
    return response.data;
  },

  getById: async (id: number): Promise<Pillow> => {
    const response = await api.get(`${ENDPOINTS.PILLOWS}/${id}`);
    return response.data;
  },

  search: async (name?: string, minPrice?: number, maxPrice?: number, page = 0, size = 12): Promise<PageResponse<Pillow>> => {
    const params: any = { page, size, sort: 'id,desc' };
    if (name) params.name = name;
    if (minPrice !== undefined) params.minPrice = minPrice;
    if (maxPrice !== undefined) params.maxPrice = maxPrice;
    const response = await api.get(ENDPOINTS.PILLOWS, { params });
    return response.data;
  },

  getSizes: async (pillowId: number): Promise<DefaultSize[]> => {
    const response = await api.get(ENDPOINTS.DEFAULT_SIZES, { params: { pillowId } });
    return response.data.content;
  },
};

File: frontend-flutter/lib/services/pillow_service.dart
import '../config/api_config.dart';
import '../models/models.dart';
import 'api_service.dart';

class PillowService {
  final _api = ApiService();

  Future<List<Pillow>> getAll({int page = 0, int size = 12}) async {
    final response = await _api.dio.get(
      ApiConfig.pillowsEndpoint,
      queryParameters: {'page': page, 'size': size, 'sort': 'id,desc'},
    );
    return (response.data['content'] as List)
        .map((json) => Pillow.fromJson(json))
        .toList();
  }

  Future<Pillow> getById(int id) async {
    final response = await _api.dio.get('${ApiConfig.pillowsEndpoint}/$id');
    return Pillow.fromJson(response.data);
  }

  Future<List<Pillow>> search(String? name, {int page = 0, int size = 12}) async {
    final params = <String, dynamic>{'page': page, 'size': size, 'sort': 'id,desc'};
    if (name != null && name.isNotEmpty) params['name'] = name;

    final response = await _api.dio.get(
      ApiConfig.pillowsEndpoint,
      queryParameters: params,
    );
    return (response.data['content'] as List)
        .map((json) => Pillow.fromJson(json))
        .toList();
  }

  Future<List<DefaultSize>> getSizes(int pillowId) async {
    final response = await _api.dio.get(
      ApiConfig.defaultSizesEndpoint,
      queryParameters: {'pillowId': pillowId},
    );
    return (response.data['content'] as List)
        .map((json) => DefaultSize.fromJson(json))
        .toList();
  }
}
